packages/rule-tester/src/RuleTester.ts

```diff
@@ -184,10 +184,37 @@ export class RuleTester extends TestFramework {
       rules: { [`${RULE_TESTER_PLUGIN_PREFIX}validate-ast`]: 'error' },
     });
 
-    this.#linter = new Linter({
-      configType: 'flat',
-      cwd: this.#testerConfig.languageOptions.parserOptions?.tsconfigRootDir,
-    });
+    // This proxy nonsense is a workaround for https://github.com/jestjs/jest/issues/14840
+    // see also https://github.com/typescript-eslint/typescript-eslint/issues/8942
+    //
+    // For some reason rethrowing exceptions skirts around the circular JSON error.
+    this.#linter = new Proxy(
```

Comment:
Can't we just wrap the `verify` call directly?

https://github.com/typescript-eslint/typescript-eslint/blob/0884b6a43c26caafb6ea7247290ca966da99cd36/packages/rule-tester/src/RuleTester.ts#L1228-L1235

Or failing that - directly override `verify`, eg
```ts
const linter = new Linter(...);
const oldVerify = linter.verify;
linter.verify = (...args) => { ... oldVerify(...args) ... };
```

Reply:
> Can't we just wrap the verify call directly?

There are three `this.#linter.verify` calls so I at least wanted to put them all in one codepath.

> Or failing that - directly override `verify`, eg
> 
> ```ts
> const linter = new Linter(...);
> const oldVerify = linter.verify;
> linter.verify = (...args) => { ... oldVerify(...args) ... };
> ```

Yeah, that works too. I just wasn't sure if `Linter` was some arcane class object that depended on crazy `this`-context and/or getter binding stuff to work correctly (like, for example, the `context` variable is within the rules) or if we were safe to do that. Can switch it to that if it seems safe.

Reply:
oh yeah now that Brad mentions it, +1 to that more direct strategy